crates/orderbook/src/database/trades.rs

```diff
@@ -25,27 +25,48 @@ pub struct TradeFilter {
 #[async_trait::async_trait]
 impl TradeRetrieving for Postgres {
     async fn trades(&self, filter: &TradeFilter) -> Result<Vec<Trade>> {
-        let _timer = super::Metrics::get()
+        let timer = super::Metrics::get()
             .database_queries
             .with_label_values(&["trades"])
             .start_timer();
 
         let mut ex = self.pool.acquire().await?;
-        database::trades::trades(
+        let trades = database::trades::trades(
             &mut ex,
             filter.owner.map(|owner| ByteArray(owner.0)).as_ref(),
             filter.order_uid.map(|uid| ByteArray(uid.0)).as_ref(),
         )
-        .map(|result| match result {
-            Ok(row) => trade_from(row),
-            Err(err) => Err(anyhow::Error::from(err)),
-        })
-        .try_collect()
-        .await
+        .map_err(anyhow::Error::from)
+        .try_collect::<Vec<TradesQueryRow>>()
+        .await?;
+        timer.stop_and_record();
+
+        let auction_order_uids = trades
+            .iter()
+            .filter_map(|t| t.auction_id.map(|auction_id| (auction_id, t.order_uid)))
+            .collect::<Vec<_>>();
+        let fee_policies = self.fee_policies(auction_order_uids.as_slice()).await?;
+
+        trades
+            .into_iter()
+            .map(|trade| {
+                let fee_policies = trade
+                    .auction_id
+                    .into_iter()
+                    .flat_map(|auction_id| {
+                        fee_policies
+                            .get(&(auction_id, trade.order_uid))
+                            .cloned()
+                            .unwrap_or_default()
+                    })
+                    .collect();
```

Comment:
nit: without additional context it's a bit confusing to `.into_iter()` on a singular thing. Maybe `.map().unwrap_or_default()` conveys that `auction_id` is optional more naturally.
Feel free to ignore, though since this is totally a matter of personal preference.

```suggestion
                let fee_policies = trade
                    .auction_id
                    .map(|auction_id| {
                        fee_policies
                            .get(&(auction_id, trade.order_uid))
                            .cloned()
                            .unwrap_or_default()
                    })
                    .unwrap_or_default();
```